### PR TITLE
Ca 91 password reset endpoints

### PR DESCRIPTION
**Pull request by JaycobDuffel**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.formatting.provider": "black"
+}
```
Comment (katiemcgoff) [inline .vscode/settings.json:3]: This file shouldn't be included

File: api/urls.py [hunk 2]
```diff
@@ -15,8 +15,11 @@
 """
 from django.contrib import admin
 from django.urls import include, path
-from rest_framework import routers
+from rest_framework import routers, generics
 from api.views import user, organization, login
+from django.contrib.auth import views as auth_views
+
+from django.conf.urls import url, include as confInclude
```
Comment (katiemcgoff) [inline api/urls.py:23]: There are some unused includes here.

File: api/models/user.py [hunk 3]
```diff
@@ -5,6 +5,11 @@
 from django.contrib.auth.models import PermissionsMixin
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
+from django.core.mail import EmailMultiAlternatives
+from django.template.loader import render_to_string
+from django.urls import reverse
+from django.dispatch import receiver
+from django_rest_passwordreset.signals import reset_password_token_created
```
Comment (katiemcgoff) [inline api/models/user.py:13]: There are some unused includes here.

File: tests/api/helpers.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    from mock import patch
+
+# try getting reverse from django.urls
+try:
+    # Django 1.10 +
+    from django.urls import reverse
+except:
+    # Django 1.8 and 1.9
+    from django.core.urlresolvers import reverse
+
```
Comment (katiemcgoff) [inline tests/api/helpers.py:16]: No need to do the exceptions here, we are using Django 3.x and Python 3 and aren't going backward.

File: tests/api/test_password_reset.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import json
+from django.test import override_settings
+from api.models.user import User
+from rest_framework import status
+from django_rest_passwordreset.models import ResetPasswordToken
+from tests.api.helpers import patch, HelperMixin
+
```
Comment (katiemcgoff) [inline tests/api/test_password_reset.py:7]: Some unused imports here.

File: tests/api/test_password_reset.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            msg="Joe should not be able to login with the old credentials"
+        )
+        self.assertTrue(
+           self.django_check_login("joe@stuff.com", 'new_secret'),
+            msg="Joe should be able to login with the new credentials"
+        )
+
+        # there should be zero tokens
+        self.assertEqual(ResetPasswordToken.objects.all().count(), 0)
+
```
Comment (katiemcgoff) [inline tests/api/test_password_reset.py:44]: Great work on these tests!

**Review by katiemcgoff — CHANGES_REQUESTED** [review]
some minor changes requested

**Review by katiemcgoff — APPROVED** [review]

**Review by DawsonDease — APPROVED** [review]

**Pull request merged**